### PR TITLE
Add preprocess script

### PR DESCRIPTION
**Pull request by maximilianpress**
This script has some extra functionality in addition to scrub_names.pl in the case of user error in naming contigs. it also works with a combined p+h fasta.

### CODE REVIEW

**Comment by bnelsj** [comment]
The script looks fine to me. Adding some tests at some point would be a good idea. Is there any reason to use the perl script over this script at this point? If not, is there documentation that should be updated?

**Comment by maximilianpress** [comment]
@bnelsj I don't see that `scrub_names.pl` should still be used, for any other reason than general conservatism. I have used this script successfully for FP in the past, though that was a while ago. I am sure that there are docs to update. The README at least mentions this. I suppose Hayley's ops guide is another thing to update.

As for testing, any testing of FP is overdue. I suppose this script is as good a place to start as any. Will try to get around to it in the next day or 2.

So I'll do these things and then merge:
1) add tests for just this script
2) update Ops guide and README

any other thoughts?

File: bin/preprocess_diploid_asm_for_fc_phase.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+def main():
+    c_args = parse_args()
+
+    if c_args["combined_fasta"] is not None:
+        p_tigs, h_tigs = parse_combined_fasta(c_args["combined_fasta"])
+    elif c_args["p_contig_fasta"] is not None and c_args["h_contig_fasta"] is not None:
+        p_tigs = parse_separated_fasta(c_args["p_contig_fasta"])
+        h_tigs = parse_separated_fasta(c_args["h_contig_fasta"])
+    else:
+        ValueError("Script requires either a combined fasta (-c arg) or both p and h fastas (-p and -h args)." \
```
Comment (bnelsj) [inline bin/preprocess_diploid_asm_for_fc_phase.py]: Need to add `raise` here.

File: bin/preprocess_diploid_asm_for_fc_phase.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    '''
+    return len(scrubbed_id.split("_")) == 2
+
+def scrub_name(fasta_id, fc_pat=FC_PAT):
+    '''remove non-falcon-unzip stuff from contig names
+    '''
+    match = re.search(fc_pat, fasta_id).group()
+    if match[-1] == "_":
+        match = match[:-1]
+    return match
```
Comment (bnelsj) [inline bin/preprocess_diploid_asm_for_fc_phase.py]: match.rstrip("_") would be a slight improvement here.

**Comment by maximilianpress** [comment]
Now have tests for the preprocessing script running in CI. Confirmed that FP gcc build is unaffected. Updated README and Hayley's ops doc.

**Pull request merged**